### ISSUE TITLE
public_suffix requires Ruby version >= 2.0

### ISSUE THREAD

**Issue by XhmikosR** [github/pages-gem#122]
Noticed this while testing #103.

https://ci.appveyor.com/project/XhmikosR/pages-gem/build/89/job/2fb60w5p4r2jffk4

/CC @benbalter @parkr @gjtorikian

**Comment by thomasjo** [comment]
Yes, Ruby 2.1.1 is what is used by Pages:
https://pages.github.com/versions/

**Comment by XhmikosR** [comment]
I'm asking in order to have a clean supported Ruby version or versions.

**Comment by thomasjo** [comment]
Can you explain what you mean by that? Not sure I understand what more you need to know; Ruby 2.1.1 is the officially required version of Ruby (as you can see on the [page I linked](https://pages.github.com/versions/)).

**Comment by XhmikosR** [comment]
https://github.com/github/pages-gem/blob/master/github-pages.gemspec#L4

**Comment by thomasjo** [comment]
Good catch! I suggest that needs to be fixed. Also noticed the `.ruby-version` file is wrong, as it specifies `2.1.0-github`.

**Comment by krmaxwell** [comment]
I've had a related [problem](https://travis-ci.org/krmaxwell/krmaxwell.github.io/builds/55669041) that Travis-CI picked up on.

**Comment by selimober** [comment]
I guess the doc for github pages needs to be updated as well: https://help.github.com/articles/using-jekyll-with-pages/

> ...and run the command ruby --version you can confirm this. Your Ruby version should begin with 1.9.3 or 2.0.0.

**Comment by G33kDude** [comment]
What @selimober said. I'm running ruby 1.9.3 and the instructions that page gave were failing to me. A bit of Googling and now I'm here.

**Comment by parkr** [comment]
> I guess the doc for github pages needs to be updated as well: https://help.github.com/articles/using-jekyll-with-pages/
>
> >...and run the command ruby --version you can confirm this. Your Ruby version should begin with 1.9.3 or 2.0.0.

@gjtorikian ^^

**Comment by benbalter** [comment]
Sounds like there's two steps here to make things actionable:

1. Could someone write up a human-readable description of the incompatibility with < 2.0? (Is it just a gemspec issue? is it an actual incompatibility, etc.)? Just trying to document the decision.

2. Once the Gem is bumped we can update the docs, but is anyone able to submit a pull request to update the Gemspec?

**Comment by XhmikosR** [comment]
@benbalter:

1. No idea, I just get the error when trying to install the pages gem with Ruby 1.9.3 (see how nice the CI is :P)
2. Sure, it's just one place so I can do it.

PS. Please see #103 last comments and ditch your PR. It's like 5 months already.

**Comment by gjtorikian** [comment]
> Could someone write up a human-readable description of the incompatibility with < 2.0?

`public_suffix`, a dependency of `github-pages-health-check`, now mandates Ruby > 2.0. This was done in a minor version bump, so `github-pages-health-check` pulled it in. I think dropping to just Ruby 2.0+ is a good idea regardless since1.9.x was end-of-lifed.

> Once the Gem is bumped we can update the docs, but is anyone able to submit a pull request to update the Gemspec?

Which gemspec? `public_suffix`? Let them stick to Ruby 2.0+ if they want to.

If we're concerned about supporting 1.9.3 (which we shouldn't, see EOL comment above), we could update `github-pages-health-check` to lock to an earlier version of `public_suffix`.

**Comment by winniehell** [comment]
Sorry, didn't intend to add a comment here...

**Comment by Zenexer** [comment]
@benbalter, the docs still need to be updated.  For example, the part about Travis CI needs to mention that the Ruby version needs to be [specified explicitly](http://docs.travis-ci.com/user/languages/ruby/#Supported-Ruby-Versions).

**Comment by peterjc** [comment]
I appreciate this is an old issue, but having just returned to working on a GitHub pages site I fell over the problem. I was able to resolve it - but only after re-reading this issue several times.

For anyone else struggling: The current TravisCI default Ruby 1.9.3 is now too old for the pages gem (see http://docs.travis-ci.com/user/languages/ruby/#Supported-Ruby-Versions ), so we should ask TravisCI to use Ruby 2.1.1 to match what is currently used on GitHub pages ( https://pages.github.com/versions/ ).

i.e. your ``.travis.yml`` should include:

```
language: ruby
rvm: 2.1.1
script: "bundle exec jekyll build"
```

The minimal ``Gemfile`` file remains just:

```
source 'https://rubygems.org'
gem 'github-pages'
```

Sadly https://help.github.com/articles/troubleshooting-github-pages-build-failures still needs to be updated.

**Comment by lrkwz** [comment]
I completely failed :-(

    $ sudo apt-get install ruby2.0 ruby2.0-dev
    $ sudo update-alternatives --install /usr/bin/ruby ruby /usr/lib/ruby/2.0.0 0
    $ sudo update-alternatives --install /usr/bin/ruby ruby /usr/lib/ruby/1.9.1 1
    $ sudo update-alternatives --config ruby

configured ruby2.0 as defeault

    $ sudo update-alternatives --display ruby
      ruby - manual mode
      link currently points to /usr/lib/ruby/2.0.0
       /usr/lib/ruby/1.9.1 - priority 1
       /usr/lib/ruby/2.0.0 - priority 0
     Current 'best' version is '/usr/lib/ruby/1.9.1'.   
     $ cat Gemfile
     source 'https://rubygems.org'
     gem 'github-pages'
     $ bundle install
     Fetching gem metadata from https://rubygems.org/............
     Fetching additional metadata from https://rubygems.org/..
     Resolving dependencies...
     (...)
     
     Gem::InstallError: public_suffix requires Ruby version >= 2.0.
    An error occurred while installing public_suffix (1.5.3), and Bundler cannot continue.
    Make sure that `gem install public_suffix -v '1.5.3'` succeeds before bundling.
    $ sudo gem install public_suffix -v '1.5.3'
    ERROR:  Error installing public_suffix:
	public_suffix requires Ruby version >= 2.0.

bump

**Comment by parkr** [comment]
@lrkwz When you run `sudo gem`, does it still use ruby2 and gem2?

**Comment by lrkwz** [comment]
Yes

    $ sudo ruby --version
    ruby 2.0.0p384 (2014-01-12) [x86_64-linux-gnu]
    $ sudo gem --version
    2.0.14

**Comment by lrkwz** [comment]
I resolved this removing all apt installed packages and using rvm

     curl -L https://get.rvm.io | bash -s stable --autolibs=enabled --ruby

**Comment by 3manuek** [comment]
@lrkwz That's the only thing that made this work for me (Linux Mint 14). For some reason it was ignoring the system ruby version.

**Comment by stefek99** [comment]
I'm doing some yak shaving now: https://twitter.com/stefek99/status/931938996896370689

Cannot do ```jekyll serve``` as ```bundle install``` is failing, due to incompatible version of ruby...

When installing ruby via ```rvm``` got some error messages suggesting `Mac 10.13`: https://pastebin.com/MyRJXX7r

**Comment by symmetriccurve** [comment]
This helped me: https://stackoverflow.com/questions/26183787/upgrading-to-ruby-2-1-3-on-mac-osx-10-9-5

**Comment by heliorcosta** [comment]
Solved: gem install public_suffix -v 1.4.6
